components/modelregistry/modelregistry.go

```diff
@@ -159,6 +162,10 @@ func (m *ModelRegistry) ReconcileComponent(ctx context.Context, cli client.Clien
 	l.Info("apply extra manifests done")
 
 	if enabled {
+		// remove leftover kube-rbac-proxy container from older ODH installs
+		if err := removeUnusedKubeRbacProxy(ctx, cli, m.GetComponentName(), dscispec.ApplicationsNamespace); err != nil {
```

Comment:
i do not understand the logic here:
1) is the problem only existing in Upgrade case:  as in ODH 2.19.0 introduced a wrong proxy ?
2) are we good with clean installation of ODH 2.20.0+ ?

why this call `removeUnusedKubeRbacProxy` is  after the deployment done , but not at the beginning or the `Reconcile()?` if it is only for the upgrade case

and is the plan( in new modelreg release) o keep modelreg deployment with  `spec/template/spec/containers/kube-rbac-proxy` in it or have it removed from manifests ?